### ISSUE TITLE
aws-cdk.aws_events: Resolution error: Supplied properties not correct for "CfnEventBusProps"

### ISSUE THREAD

**Issue by ashleydaley** [aws/aws-cdk#17172]
### What is the problem?

After pip install aws-cdk 1.129.0, when creating a Stack that contains an `aws_events.EventBus` that does not have (optional) `event_bus_name` defined errors with the following message when we `cdk synth` the app:

```
jsii.errors.JavaScriptError: 
  Error: Resolution error: Supplied properties not correct for "CfnEventBusProps"
    name: required but missing.
      at ValidationResult.assertSuccess (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/runtime.js:139:19)
      at cfnEventBusPropsToCloudFormation (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/aws-events/lib/events.generated.js:476:43)
      at CfnEventBus.renderProperties (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/aws-events/lib/events.generated.js:560:16)
      at PostResolveToken.ret.Resources [as processor] (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/cfn-resource.js:303:52)
      at PostResolveToken.postProcess (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/util.js:81:21)
      at Object.postProcess (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:59:82)
      at DefaultTokenResolver.resolveToken (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/resolvable.js:59:38)
      at resolve (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:130:55)
      at Object.resolve (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:57:48)
      at resolve (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:145:51)
```

After downgrading python module to aws-cdk 1.128.0, the same error does not occur and stack is built successfully

### Reproduction Steps

```
import os
from aws_cdk import core, aws_events

env = core.Environment(region=os.environ["AWS_DEFAULT_REGION"], account=os.environ["AWS_ACCOUNT_ID"])


class TestStack(core.Stack):
    def __init__(
        self,
        scope: core.Construct,
        id: str,
        **kwargs,
    ):
        core.Stack.__init__(self, scope, id, **kwargs)
        aws_events.EventBus(self, "event_bus")


app = core.App()
test_stack = TestStack(app, "test-stack", env=env)
app.synth()

```

### What did you expect to happen?

The stack should synth and build normally, and an auto-generated event_bus_name should be created

### What actually happened?

The synth failed with

````
jsii.errors.JavaScriptError: 
  Error: Resolution error: Supplied properties not correct for "CfnEventBusProps"
    name: required but missing.
      at ValidationResult.assertSuccess (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/runtime.js:139:19)
      at cfnEventBusPropsToCloudFormation (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/aws-events/lib/events.generated.js:476:43)
      at CfnEventBus.renderProperties (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/aws-events/lib/events.generated.js:560:16)
      at PostResolveToken.ret.Resources [as processor] (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/cfn-resource.js:303:52)
      at PostResolveToken.postProcess (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/util.js:81:21)
      at Object.postProcess (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:59:82)
      at DefaultTokenResolver.resolveToken (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/resolvable.js:59:38)
      at resolve (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:130:55)
      at Object.resolve (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:57:48)
      at resolve (/tmp/jsii-kernel-RpWQUL/node_modules/@aws-cdk/core/lib/private/resolve.js:145:51)
````

### CDK CLI Version

1.129.0 (build fb43f89)

### Node.js Version

14.7.5

### OS

Amazon Linux 2

### Language

Python

### Language Version

3.8.8

**Comment by gavllew** [comment]
I think this had something to do with fix #17008 for issue #14337.  Until then, CDK was already inventing a name on-the-fly.

**Comment by jasonwadsworth** [comment]
As a work around I ended up setting the name using the name that was generated in previous versions of the CDK.
